### PR TITLE
Pathmapping movement imperative now works correctly with Single Step …

### PR DESCRIPTION
**Pull request by EvidentlyCube**
…+ updated help

[Relevant thread](http://forum.caravelgames.com/viewtopic.php?TopicID=42309)

### CODE REVIEW

**Review by mrimer — APPROVED** [review]
It's good!

I have a question on the comments' text.

File: DRODLib/Character.cpp [hunk 1]
```diff
@@ -1352,9 +1352,10 @@ void CCharacter::Process(
 				bool bPathmapping;
 				const bool bAllowTurning = !pw;
 				const bool bStopTurn = GetMovement(wDestX, wDestY, dx, dy, dxFirst, dyFirst, bPathmapping, bAllowTurning);
-				if (bStopTurn)
+				//Even if pathfinding move says to wait until next turn ignore if Single Step is on
```
[the remark this answers is not in the document]
Comment (EvidentlyCube) [reply DRODLib/Character.cpp:1355]: I was very confused by the value returned by `GetMovement()`, which is why once I understood what it does to comment it a bit more thoroughly.

How about this?
```
//When pathfinding indicates to not move here, 'Single Step' should still cause the script to move on on the next turn
```

File: DRODLib/Character.cpp [hunk 2]
```diff
@@ -1352,9 +1352,10 @@ void CCharacter::Process(
...
+				if (bStopTurn && !ph)
 					STOP_COMMAND;
-				if (!dx && !dy)
+				if (bStopTurn || (!dx && !dy))
```
[the remark this answers is not in the document]
Comment (EvidentlyCube) [reply DRODLib/Character.cpp:1358]: Yup. I already implemented it and realized "waitasec, something wrong here" before I came back  to see your edit.

File: DRODLib/Character.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrimer) [reply DRODLib/Character.cpp:1355]: How about:

[the remark this answers is not in the document]
Comment (mrimer) [reply DRODLib/Character.cpp:1355]: There seems to be some ambiguity in that verbiage.  How about:
`//When pathfinding indicates to not move, 'Single Step' causes script execution to advance on the next turn.`

**Review by mrimer — APPROVED** [review]

**Pull request merged**